src/main/java/org/openstreetmap/atlas/checks/validation/tag/BridgeDetailedInfoCheck.java

```diff
@@ -0,0 +1,108 @@
+package org.openstreetmap.atlas.checks.validation.tag;
+
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Optional;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.BridgeTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.RailwayTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+
+/**
+ * Flags railway and major highway bridges which are longer than configured minimum and have
+ * unspecified structure. This is a port of Osmose check 7012.
+ * 
+ * @author ladwlo
+ */
+public class BridgeDetailedInfoCheck extends BaseCheck<Long>
+{
+
+    private static final long serialVersionUID = -8915653487119336836L;
+
+    private static final EnumSet<HighwayTag> MAJOR_HIGHWAYS = EnumSet.of(HighwayTag.MOTORWAY,
+            HighwayTag.TRUNK, HighwayTag.PRIMARY, HighwayTag.SECONDARY);
+    private static final Double MINIMUM_LENGTH = 500.0;
+    public static final String BRIDGE_STRUCTURE_TAG = "bridge:structure";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
```

Comment:
nit: Since there is only a single instruction, you can use Collections.singletonList instead of Arrays.asList.